src/components/ScrollWheel/Knob.tsx

```diff
@@ -0,0 +1,415 @@
+import React, { SyntheticEvent } from 'react';
+import styled from 'styled-components';
+import { useRef } from 'react';
+import { useEffectOnce } from 'hooks';
+import { useEffect } from 'react';
+
+const Container = styled.div`
+  user-select: none;
+  position: relative;
+  display: flex;
+  justify-content: center;
+  margin: auto 0;
+  touch-action: none;
+  transform: translate3d(0, 0, 0);
+`;
+
+const CanvasContainer = styled.div<{ width: number, height: number }>`
+  position: relative;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  width: ${(props) => props.width}px;
+  height: ${(props) => props.height}px;
+`;
+
+const Canvas = styled.canvas`
+  border-radius: 50%;
+  border: 1px solid #b9b9b9;
+  background: white;
+`;
+
+const CenterButton = styled.div<{ size: number }>`
+  position: absolute;
+  top: 0;
+  bottom: 0;
+  left: 0;
+  right: 0;
+  margin: auto;
+  width: ${(props) => props.size / 2.5}px;
+  height: ${(props) => props.size / 2.5}px;
+  border-radius: 50%;
+  box-shadow: rgb(191, 191, 191) 0px 1em 3em inset;
+  background: rgb(225, 225, 225);
+  border: 1px solid #b9b9b9;
+
+  :active {
+    filter: brightness(0.9);
+  }
+`;
+
+const WheelButton = styled.img<{
+  margin?: string,
+  top?: string,
+  bottom?: string,
+  left?: string,
+  right?: string
+}
+  >`
+  position: absolute;
+  margin: ${(props) => props.margin};
+  top: ${(props) => props.top};
+  bottom: ${(props) => props.bottom};
+  left: ${(props) => props.left};
+  right: ${(props) => props.right};
+  user-select: none;
+  pointer-events: none;
+  max-height: 13px;
+`;
+
+/** Custom Event from https://github.com/john-doherty/long-press-event  */
+interface LongPressEvent extends SyntheticEvent {
+  detail: {
+    clientX: number,
+    clientY: number
+  }
+}
+
+const ANGLE_ARC = (360 * Math.PI) / 180;
+const ANGLE_OFFSET = (0 * Math.PI) / 180;
+const START_ANGLE = 1.5 * Math.PI + ANGLE_OFFSET;
+const END_ANGLE = 1.5 * Math.PI + ANGLE_OFFSET + ANGLE_ARC;
```

Comment:
These constants are all based on the previous code. So some of them might be pretty redundant -- I wanted to mostly preserve the existing logic, since the file is quite large